File: app/src/test/java/moshi/PersonJsonAdapterTest.java
package moshi;

import static org.assertj.core.api.Assertions.assertThat;

import com.squareup.moshi.JsonAdapter;
import com.squareup.moshi.Moshi;

import org.junit.Before;
import org.junit.Test;

import java.io.IOException;

public class PersonJsonAdapterTest {

    String personJson = "{"
            + "\"age\":21,"
            + "\"gender\":\"female\","
            + "\"name\":\"Kate\""
            + "}";

    Person person = new Person("Kate", 21, Gender.female);

    private Moshi moshi;
    private JsonAdapter<Person> adapterPerson;

    @Before
    public void setUp() {
        moshi = new Moshi.Builder().build();
        adapterPerson = moshi.adapter(Person.class);
    }

    @SuppressWarnings("ConstantConditions")
    @Test
    public void fromJson() throws IOException {
        Person person = adapterPerson.fromJson(personJson);

        assertThat(person.getName()).isEqualTo("Kate");
        assertThat(person.getAge()).isEqualTo(21);
        assertThat(person.getGender()).isEqualTo(Gender.female);
    }

    @Test
    public void toJson() {
        String result = adapterPerson.toJson(person);

        assertThat(result).isEqualTo(personJson);
    }

}

File: app/src/test/java/moshi/PersonJsonAdapterTest.kt
package moshi

import com.squareup.moshi.JsonAdapter
import com.squareup.moshi.Moshi
import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test

class PersonJsonAdapterTest {

    private var personJson = ("{"
            + "\"age\":21,"
            + "\"gender\":\"female\","
            + "\"name\":\"Kate\""
            + "}")

    private var person = Person("Kate", 21, Gender.female)

    private lateinit var moshi: Moshi
    private lateinit var adapterPerson: JsonAdapter<Person>

    @Before
    fun setUp() {
        moshi = Moshi.Builder().build()
        adapterPerson = moshi.adapter(Person::class.java)
    }

    @Test
    fun fromJson() {
        val person = adapterPerson.fromJson(personJson)

        assertThat(person!!.name).isEqualTo("Kate")
        assertThat(person.age).isEqualTo(21)
        assertThat(person.gender as Gender).isEqualTo(Gender.female)
    }

    @Test
    fun toJson() {
        val result = adapterPerson.toJson(person)

        assertThat(result).isEqualTo(personJson)
    }

}